pkg/service/types.go

```diff
@@ -43,9 +48,60 @@ type MeshService struct {
 
 	// Protocol is the protocol served by the service's port
 	Protocol string
+
+	providerKey string
+
+	subdomain string
+
+	subdomainPopulated bool
+}
+
+// NamespacedKey is the key (i.e. namespace + name) with which to lookup the backing service within the provider
+func (ms MeshService) NamespacedKey() string {
+	return fmt.Sprintf("%s/%s", ms.Namespace, ms.ProviderKey())
```

Comment:
The comment indicates it should be `namespace/name`, while the code uses `namespace/providerKey`.  This function should either be renamed or the comment needs to be updated.

Reply:
Good catch!